lib/spack/spack/util/shell_detection.py

```diff
@@ -0,0 +1,35 @@
+# Copyright 2013-2024 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+import os
+import platform
+import subprocess
+
+from spack.error import SpackError
+
+
+def active_shell_type(env=os.environ):
+    if platform.system() == "Windows":
+        if "POWERSHELL" in env:
+            return "ps1"
+        else:
+            try:
+                output = subprocess.check_output(
+                    'powershell -Command "echo $PSVersionTable"', universal_newlines=True
```

Comment:
This will always pass (fine if we don't want to support CMD), as the powershell executable is always in the path unless a user removed it explicitly. 

I think something like:
```
pproc_name = psutil.Process(os.getppid()).name()
is_power_shell = bool(re.fullmatch('pwsh|pwsh.exe|powershell|powershell.exe', pproc_name))
```

With a better regex would be best.